d3a_api_client/websocket_device.py

```diff
@@ -0,0 +1,79 @@
+import asyncio
+import logging
+import json
+import threading
+import traceback
+import websockets
+from d3a_interface.utils import wait_until_timeout_blocking
+
+
+class RestWebsocketAPIException(Exception):
+    pass
+
+
+class WebsocketMessageReceiver:
+    def __init__(self, rest_client):
+        self.client = rest_client
+        self.command_response_buffer = []
+
+    def received_message(self, message):
+        if "event" in message:
+            if message["event"] == "market":
```

Comment:
Shouldn't we introduce an Enum at this point as we will exactly have only three type of events ?

Reply:
Well, this is a websocket message, and we receive it from the network connection, therefore in order to expose an Enum here it has to be populated in d3a. Therefore d3a-interface has to be involved in order to keep the enum definition and be reused across d3a and d3a-api-client. Not negative to introduce it, but not for now. Maybe after the deploy.